### PR TITLE
Can optimization

### PR DESCRIPTION
**Pull request by mflanag**
I have made motor commands be sent on a separate can bus. It has been confirmed through testing on a full session. I changed the main loop to run at 50 hz and used a vtaskdelayuntil to ensure the frequency. I also changed how we do motor controller polling to do it on the separate bus.

This code will work for the future when we have a separate canbus for the drive critical functions.

Need to add protections around variables to ensure read while writing does not occur (new feature)

### CODE REVIEW

**Comment by RyanSchw** [comment]
Are we calling it CAN1 and CAN2 or DCAN and VCAN? I know @bsmrdel had a message where it was referenced as DCAN and VCAN on the PCBs.

**Comment by mflanag** [comment]
I can change it to be consistent with Brad. I finished writing this before he wrote that.

**Comment by mflanag** [comment]
i changed it

**Comment by mflanag** [comment]
I changed it

**Comment by RyanSchw** [comment]
So that’s the one we want to stick with? I don’t have an issue either way... just want to standardize.

________________________________

**Comment by mflanag** [comment]
yeah brad's is better (you know what it is just by reading it)

File: Src/car.c [hunk 1]
```diff
@@ -167,22 +167,22 @@ void initRTOSObjects() {
...
-	car.m_CAN =				xSemaphoreCreateMutex(); //mutex to protect CAN peripheral
+	car.q_mc_frame = 		xQueueCreate(QUEUE_SIZE_MCFRAME, sizeof(CanRxMsgTypeDef));
 
 	/* Create Tasks */
 
 	//todo optimize stack depths http://www.freertos.org/FAQMem.html#StackSize
 	xTaskCreate(taskPedalBoxMsgHandler, "PedalBoxMsgHandler", 256, NULL, 1, NULL);
 	xTaskCreate(taskCarMainRoutine, "CarMain", 256 , NULL, 1, NULL);
-	xTaskCreate(taskTXCAN, "TX CAN", 256, NULL, 1, NULL);
+	xTaskCreate(taskTX_DCAN, "TX CAN 1", 256, NULL, 1, NULL);
```
Comment (RyanSchw) [inline Src/car.c]: Shouldn't the string be `"TX DCAN"`?

File: Inc/car.h [hunk 2]
```diff
@@ -146,19 +147,21 @@ typedef struct {
...
-
-	CAN_HandleTypeDef *		phcan;						//pointer to car's CAN peripheral handle
+	CAN_HandleTypeDef *		phdcan;						//pointer to car's CAN peripheral handle
+	CAN_HandleTypeDef *		phvcan;
 
 } Car_t;
 
 extern volatile Car_t car;
 extern CAN_HandleTypeDef hcan1;
+extern CAN_HandleTypeDef hcan2;
```
[the remark this answers is not in the document]
Comment (RyanSchw) [reply Inc/car.h]: I may have been mistaken... was this automatically generated by CubeMX? If so, then that's fine. There's nothing we can do about that.

**Review by RyanSchw — COMMENTED** [review]
I added a few comments about changing to `DCAN` and `VCAN`, however I'm just now realizing the lines I commented may be generated by CubeMX and not touched by the user. We may benefit from adding something like `dcan = hcan1` and just using `dcan` from that point.

File: Inc/CANProcess.h [hunk 3]
```diff
@@ -17,7 +17,7 @@
 #define ID_PEDALBOX2							0x501
 #define	ID_PEDALBOXCALIBRATE					0x503
 #define ID_BAMOCAR_STATION_TX					0x210	//message recieved by MC
-#define ID_BAMOCAR_STATION_RX					0x190	//message sent by MC
+#define ID_BAMOCAR_STATION_RX					0x180	//message sent by MC
```
[the remark this answers is not in the document]
Comment (mflanag) [reply Inc/CANProcess.h]: it was wrong.

File: Inc/car.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (mflanag) [reply Inc/car.h]: these are cube generated

File: Src/CANProcess.c [hunk 4]
```diff
@@ -43,28 +43,24 @@
 ***************************************************************************/
 void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef *hcan)
 {
-	HAL_GPIO_TogglePin(LD4_GPIO_Port, LD4_Pin);
+	//HAL_GPIO_TogglePin(LD4_GPIO_Port, LD4_Pin);
```
[the remark this answers is not in the document]
Comment (mflanag) [reply Src/CANProcess.c]: because im using it in other places

File: Src/car.c [hunk 5]
```diff
@@ -47,7 +46,8 @@ void carInit() {
 	car.pb_mode = PEDALBOX_MODE_DIGITAL;
 	car.throttle_acc = 0;
 	car.brake = 0;
-	car.phcan = &hcan1;
+	car.phdcan = &hcan1;
```
[the remark this answers is not in the document]
Comment (mflanag) [reply Src/car.c]: no

**Comment by mflanag** [comment]
I have updated the code

**Comment by mflanag** [comment]
I think i have complied with everything can we merge?

**Review by RyanSchw — APPROVED** [review]

**Pull request merged**